redwoods-insurance-salesforce/force-app/main/default/lwc/incidentImageCarousel/__tests__/incidentImageCarousel.test.js

```diff
@@ -19,4 +32,24 @@ describe('c-incident-image-carousel', () => {
         const labelElement = element.shadowRoot.querySelector('p');
         expect(labelElement.textContent).toContain('no images');
     });
+
+    it('renders a picture when one is present', () => {
+        // Emit data from @wire
+        getRecordAdapter.emit('mockRecordId');
+        getRelatedPicturesAdapter.emit(mockTwoImages);
+
+        // Create initial element
+        const element = createElement('c-incident-image-carousel', {
+            is: IncidentImageCarousel
+        });
+        document.body.appendChild(element);
+
+        return Promise.resolve().then(() => {
+            const imageEls = element.shadowRoot.querySelectorAll(
+                'lightning-carousel-image'
+            );
+            console.log(JSON.stringify(imageEls));
+            expect(imageEls).not.toBe(null);
```

Comment:
As for that the test passes, that is correct. `querySelectorAll` returns an empty `NodeList` when no matches are found, so it's never null.

You've to check for the length, i. e. `expect(imageEls.length).toEqual(mockTwoImages.length);`

Reply:
Check for length, gotcha. I can do that.

What I don't understand is why it's returning {}. I thought it was a race condition, so I added the whole return Promise.resolve().then trick, but that's still not working as I epxect.